crates/sncast/src/lib.rs

```diff
@@ -269,63 +286,92 @@ fn get_account_from_keystore<'a>(
         bail!("File containing the account does not exist: When using `--keystore` argument, the `--account` argument should be a path to the starkli JSON account file");
     }
 
-    let signer = LocalWallet::from(SigningKey::from_keystore(
+    let private_key = SigningKey::from_keystore(
         keystore_path,
         get_keystore_password(KEYSTORE_PASSWORD_ENV_VAR)?.as_str(),
-    )?);
+    )?
+    .secret_scalar();
+    let private_key = format!("{private_key:#x}");
 
     let file_content = fs::read_to_string(path_to_account.clone())
         .with_context(|| format!("Failed to read a file = {}", &path_to_account))?;
     let account_info: Value = serde_json::from_str(&file_content)
         .with_context(|| format!("Failed to parse file = {} to JSON", &path_to_account))?;
-    let address = FieldElement::from_hex_be(
-        account_info
-            .get("deployment")
-            .and_then(|deployment| deployment.get("address"))
-            .and_then(Value::as_str)
-            .ok_or_else(|| anyhow::anyhow!("Failed to get address from account JSON file - make sure the account is deployed"))?
-    )?;
-
-    Ok(SingleOwnerAccount::new(
-        provider,
-        signer,
+    let deployment_info = account_info
+        .get("deployment")
+        .expect("Failed to find deployment key in the account JSON file");
+
+    let parse_to_string = |entry: &Value, key: &str| -> Option<String> {
+        entry.get(key).and_then(Value::as_str).map(str::to_string)
+    };
+
+    let address = parse_to_string(deployment_info, "address").ok_or_else(|| {
+        anyhow::anyhow!(
+            "Failed to get address from account JSON file - make sure the account is deployed"
+        )
+    })?;
+    let class_hash = parse_to_string(deployment_info, "class_hash");
+    let salt = parse_to_string(deployment_info, "salt");
+    let deployed = parse_to_string(deployment_info, "status").map(|status| &status == "deployed");
+
+    let variant_info = account_info
+        .get("variant")
+        .expect("Failed to find variant key in the account JSON file");
+    let public_key = parse_to_string(variant_info, "public_key")
+        .ok_or_else(|| anyhow::anyhow!("Failed to get public key from account JSON file"))?;
+    let legacy = variant_info.get("legacy").and_then(Value::as_bool);
+
+    Ok(Account {
+        private_key,
+        public_key,
         address,
-        chain_id,
-        ExecutionEncoding::Legacy,
-    ))
+        salt,
+        deployed,
+        class_hash,
+        legacy,
+    })
 }
 
-fn get_account_from_accounts_file<'a>(
```

Comment:
Note: `get_account_info_from_accounts_file` and `get_account_info_from_keystore` can also be utilized within `starknet_commands/account/deploy.rs` to refactor and simplify the code.